File: ChristmasJoy.App/Models/UserStatus.cs
namespace ChristmasJoy.App.Models
{
  public enum Status
  {
    Grinch = 0,
    Snowman = 1,
    Elf = 2,
    Cookie = 3,
    Santa = 4,
    Magus = 5
  }

  public class UserStatus
  {
    public int Id { get; set; }

    public string UserName { get; set; }

    public string ChristmasStatus { get; set; }

    public double Points { get; set; }

    public int NoOfComments { get; set; }
  }
}

File: ChristmasJoy.App/src/app/models/UserStatus.ts
import { StatusPoints } from './StatusPoints';

export class UserStatus{
    userName: string;
    christmasStatus: string;
    points: number;
    noOfComments: number;
    id: number;

    public static nextStatusData(christmasStatus, points): any{
        let statusIndex = -1;
        let pointsMap = StatusPoints.map;
        for(var i=0; i< pointsMap.length; i++){
            if(pointsMap[i].Status == christmasStatus){
                statusIndex = i;
            }
        }
        if(statusIndex > -1 && statusIndex < pointsMap.length -1){
            let nextPoints = pointsMap[statusIndex +1].Points;
            return {
                Status: pointsMap[statusIndex +1].Status,
                Percentage: Math.round((points * 100)/nextPoints)
            };
        }
        return null;
    }
}
